R/board_url.R

```diff
@@ -69,14 +112,19 @@ pin_exists.pins_board_url <- function(board, name, ...) {
 pin_meta.pins_board_url <- function(board, name, version = NULL, ...) {
   check_name(name)
   check_pin_exists(board, name)
-
-  if (!is.null(version)) {
-    abort("board_url() doesn't support versions")
+  if (!is.null(version) && !board$versioned) {
+    abort_board_not_versioned("board_url")
   }
 
   url <- board$urls[[name]]
-  is_dir <- grepl("/$", url)
 
+  if (board$versioned) {
+    versions <- pin_versions(board, name)$version
+    version <- check_pin_version(board, name, version)
+    url <- board$urls[[name]][[which(versions == version)]]
```

Comment:
Why `which()` here?

Reply:
To choose the right URL when there is more than one version:

```r
Browse[2]> versions <- pin_versions(board, name)$version
Browse[2]> versions
[1] "20221215T205813Z-5026d" "20221215T205813Z-cba09"
Browse[2]> version <- check_pin_version(board, name, version)
Browse[2]> version
[1] "20221215T205813Z-cba09"
Browse[2]> board$urls[[name]]
[1] "http://127.0.0.1:64269/y/20221215T205813Z-5026d/" "http://127.0.0.1:64269/y/20221215T205813Z-cba09/"
Browse[2]> board$urls[[name]][[which(versions == version)]]
[1] "http://127.0.0.1:64269/y/20221215T205813Z-cba09/"
```

Let me know if I am misunderstanding the question!

Reply:
Why doesn't `versions == version` alone work?

Reply:
Oh gosh, yep -- thanks!